pkg/api/types.go

```diff
@@ -1205,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
```

Comment:
That's correct, we could use the NodeName directly, and we can do that for now to get something working quickly.   But if we want to do something more generic, and have it work for any arbitrary topology like zones, racks, etc, then the current idea is to use the label approach, similar to how pod affinity/anti-affinity works.